### PR TITLE
Adding Index Generation for Relationships and Standalone Fields

### PR DESCRIPTION
**Pull request by jcora-nyt**
I'll send you guys the _composer.json_ file to use to test this feature locally.

### CODE REVIEW

File: res/code_templates/index/custom/Extension/modules/__module__/Ext/Vardefs/idx___index__.php.twig [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ * @author SugarCli
+ * SugarCli created file
+ */
+
+// {{ index }} index definition
+$dictionary["{{ moduleBean }}"]["indices"][] = array(
+    // Basic index features
+    'name' => 'idx_{{ index }}',
+    'type' => 'index',
+    'fields' => array(
```
Comment (ashutoshsinghnyt) [inline res/code_templates/index/custom/Extension/modules/__module__/Ext/Vardefs/idx___index__.php.twig:12]: We may want to define type for some of the indices.

File: res/code_templates/relationship-right/custom/Extension/modules/__module__/Ext/Vardefs/idx_for___relationship-right__.php.twig [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?php
+/**
+ * @author SugarCli
+ * SugarCli created file
+ */
+
+// Relationship foreign key to {{ moduleLeft|replace({'_': ' '}) }} index
+$dictionary["{{ moduleBean }}"]["indices"][] = array(
```
Comment (ashutoshsinghnyt) [inline res/code_templates/relationship-right/custom/Extension/modules/__module__/Ext/Vardefs/idx_for___relationship-right__.php.twig:8]: - Should we consider ability to add length for an index as optional parameter from CLI tool or it should be a manual job by tweaking the index definition file?
- Tried to define index on multiple columns but the command is putting encoded string in the fields key of index definition.

**Review by ashutoshsinghnyt — CHANGES_REQUESTED** [review]

File: res/code_templates/index/custom/Extension/modules/__module__/Ext/Vardefs/idx___index__.php.twig (hunk 1 above)
Comment (jcora-nyt) [reply res/code_templates/index/custom/Extension/modules/__module__/Ext/Vardefs/idx___index__.php.twig:12]: I'm not sure what you mean. There is only one index type.

File: res/code_templates/relationship-right/custom/Extension/modules/__module__/Ext/Vardefs/idx_for___relationship-right__.php.twig (hunk 2 above)
Comment (jcora-nyt) [reply res/code_templates/relationship-right/custom/Extension/modules/__module__/Ext/Vardefs/idx_for___relationship-right__.php.twig:8]: In Sugar 7.7 I don't see length as an available option for a index. Regardless, I see it as being a manual step similar to change varchar length for fields.

Comment (jcora-nyt) [reply res/code_templates/relationship-right/custom/Extension/modules/__module__/Ext/Vardefs/idx_for___relationship-right__.php.twig:8]: I see the encoding issue you mentioned (weird), and I'll get a fix in for it.

**Comment by jcora-nyt** [comment]
@ashutoshsinghnyt: Encoding problem for multiples fields was fixed. Please continue the review.

File: res/code_templates/index/custom/Extension/modules/__module__/Ext/Vardefs/idx___index__.php.twig (hunk 1 above)
Comment (ashutoshsinghnyt) [reply res/code_templates/index/custom/Extension/modules/__module__/Ext/Vardefs/idx___index__.php.twig:12]: Other types are  **_Primary_** and _**Unique**_.

**Comment by ashutoshsinghnyt** [comment]
Looks Good, Merging.

**Pull request merged**